core/trino-spi/src/test/java/io/trino/spi/TestSpiBackwardCompatibility.java

```diff
@@ -79,7 +79,7 @@ public void testBackwardIncompatibleEntitiesAreInPreviousSpi()
 
     private static Set<String> getBackwardIncompatibleChanges()
     {
-        String version = new ConnectorContext() {}.getSpiVersion().replace("-SNAPSHOT", "");
+        String version = new ConnectorContext() {}.getSpiVersion().substring(0, 3);
```

Comment:
... and this doesn't look like "obviously correct". Why is it correct thing to do?

Reply:
This is not a good assumption. I suspect it will break for anyone that maintains a fork of Trino and a custom versioning scheme.

Reply:
I think the current code, before the change, breaks anyone who maintains a fork of Trino and a custom versioning scheme. The test has exclusions (allowed breaking changes) and it needs to map "current version" to the versions used in the exclusions. The exclusions use Trino versioning, so the version extraction needs to accommodate for that. Of course, this is not possible for _any_ custom versioning, but it's possible for any versioning that's based on Trino version, which is a reasonable assumption.

@kokosing  please change this to more defensive approach, to take care of hypothetical forks we can be concerned about:
```
String majorVersion = new ConnectorContext() {}.getSpiVersion().replace("^(\\d+)([-.].*)$", "$1");
```